pysr/regressor_sequence.py

```diff
@@ -0,0 +1,381 @@
+            X.flatten(), self.recursive_history_length * np.prod(X.shape[1])
+        )[:: X.shape[1], :]
+        pred = self._regressor.predict(X=historical_X, index=index)
+        if extra_predictions > 0:
+            output = pred
+            previous_points = historical_X[-1]
+            # Without this, the model will re-predict the last data point
+            pred_once = self._regressor.predict(X=[previous_points], index=index)
+            previous_points = previous_points[X.shape[1] :]
+            previous_points = np.append(previous_points, pred_once)
+            previous_points = previous_points.flatten()
+            for _ in range(extra_predictions):
+                pred_once = self._regressor.predict(X=[previous_points], index=index)
+                previous_points = previous_points[X.shape[1] :]
+                previous_points = np.append(previous_points, pred_once)
+                previous_points = previous_points.flatten()
+                output = np.append(output, pred_once)
+            return output.reshape(-1, X.shape[1])
+        return pred
+
+    def from_file(
+        self,
+        cls,
+        equation_file: PathLike,
+        *pysr_args,
+        binary_operators: Optional[List[str]] = None,
+        unary_operators: Optional[List[str]] = None,
+        n_features_in: Optional[int] = None,
+        feature_names_in: Optional[ArrayLike[str]] = None,
+        selection_mask: Optional[NDArray[np.bool_]] = None,
+        nout: int = 1,
+        **pysr_kwargs
+    ):
+        return self._regressor.from_file(
+            cls,
+            equation_file,
+            *pysr_args,
+            binary_operators,
+            unary_operators,
+            n_features_in,
+            feature_names_in,
+            selection_mask,
+            nout,
+            **pysr_kwargs,
+        )
+    
+    def __repr__(self):
+        return self._regressor.__repr__()
+
+    def __getstate__(self):
+        return self._regressor.__getstate__()
+    
+    @property
+    def julia_options_(self):
+        return self._regressor.julia_options_
+
+    @property
+    def julia_state_(self):
+        return self._regressor.julia_state_
+    
+    def get_best(self, index=None):
+        return self._regressor.get_best(index=index)
+    
+    def refresh(self, checkpoint_file: Optional[PathLike] = None) -> None:
+        return self._regressor.refresh(checkpoint_file=checkpoint_file)
+    
+    def sympy(self, index=None):
+        return self._regressor.sympy(index=index)
+    
+    def latex(self, index=None, precision=3):
+        return self._regressor.latex(index=index, precision=precision)
+    
+    def get_hof(self):
+        return self._regressor.get_hof()
+    
+    def latex_table(
+        self,
+        indices=None,
+        precision=3,
+        columns=["equation", "complexity", "loss", "score"],
+    ):
+        return self._regressor.latex_table(indices=indices, precision=precision, columns=columns)
+    
```

Comment:
These are incorrect since the latex has `y=...`. But `y` is not defined in the sequence regressor